message-queue-lib/test/cmr/message_queue/test/queue/sqs.clj

```diff
@@ -54,26 +74,74 @@
   (proxy [GetQueueUrlResult]
          []
          (getQueueUrl [] url-str)))
-                         
-(defn- client-proxy
-  "Creates a proxy for AWSSQSClient. The proxy returns a GetQueueUrlRequest
-  proxy for calls to getQueueUrl. For calls to receiveMessages it calls the
-  functions in receive-fns starting with the first function and then using 
+
+(defn- recv-fn
+  "Returns a function to call on receive messages."
+  [receive-fns idx]
+  (if (< idx (count receive-fns))
+    (nth receive-fns idx)
+    #(throw (ExitException.))))
+
+(defn- fake-client-proxy
+  "Creates a proxy for AWSSQSClient. This prox does the following:
+
+  * Returns a `GetQueueUrlRequest` proxy for calls to getQueueUrl.
+  * For calls to `receiveMessages`, it calls the functions in receive-fns
+    starting with the first function and then using the next function in the
+    list for each subsequent call until the list of receive functions is
+    exhausted. At that point subsequent calls to receiveMessages will throw an
+    ExitException to force the calling thread to exit. This is done to prevent
+    test worker threads from persisting after tests complete."
```

Comment:
Thank you. I was actually looking to do exactly this the last time I ran the AWS tests, but ran out of time to figure out how to do it.

Reply:
🙇